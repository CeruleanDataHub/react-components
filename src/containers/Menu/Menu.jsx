import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Icon } from "../Icon/Icon";
import { Typography } from "../Typography/Typography";

const MenuIcon = styled.span`
  font-size: 1.5em;
`;

const MenuItems = styled.ul`
  margin: 1em 0 0 0.25em;
  padding: 0;
  list-style-type: none;
`;

const MenuItem = styled.li`
  display: block;
  padding: 1em;
  line-height: 1.5em;
  height: 1.5em;
  font-weight: 300;
  text-transform: uppercase;

  a {
    text-decoration: none;
    color: inherit;
  }
`;

const Flex = styled.div`
  display: flex;
`;

const MenuItemText = styled.span`
  margin-left: 1rem;
  /* text still selectable in browser */
  opacity: ${({ menuOpen }) => (menuOpen ? 1 : 0)};
`;

const SingleItem = ({ name, icon, LinkComponent, to, menuOpen }) => (
  <MenuItem>
    <Typography fontFamily="exo">
      <LinkComponent to={to}>
        <Flex>
          <Icon name={icon} as={MenuIcon} />
          <MenuItemText className="menu-text" menuOpen={menuOpen}>
            {name}
          </MenuItemText>
        </Flex>
      </LinkComponent>
    </Typography>
  </MenuItem>
);

SingleItem.propTypes = {
  name: PropTypes.string,
  icon: Icon.propTypes.name,
  to: PropTypes.string,
  LinkComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  menuOpen: PropTypes.bool
};

const DefaultLinkComponent = ({ children }) => children;
DefaultLinkComponent.propTypes = {
  children: PropTypes.node
};

DefaultLinkComponent.defaultProps = {
  children: ""
};

SingleItem.defaultProps = {
  name: "",
  icon: "",
  to: "",
  LinkComponent: DefaultLinkComponent,
  menuOpen: true
};

export const Menu = ({ items, menuOpen }) => (
  <MenuItems>
    {items.map(({ name, icon, LinkComponent, to }) => (
      <SingleItem
        name={name}
        icon={icon}
        LinkComponent={LinkComponent}
        to={to}
        menuOpen={menuOpen}
        key={name}
        data-menu-item-test
      />
    ))}
  </MenuItems>
);

Menu.propTypes = {
  /** Menu items to be rendered */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      icon: Icon.propTypes.name,
      to: PropTypes.string,
      LinkComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
    })
  ),
  /** Defines whether the menu should be open or closed */
  menuOpen: PropTypes.bool
};

Menu.defaultProps = {
  items: [],
  menuOpen: true
};
