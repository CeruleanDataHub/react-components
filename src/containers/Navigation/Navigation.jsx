import PropTypes from "prop-types";
import React, { useState } from "react";
import styled from "styled-components";

import { Icon } from "../Icon/Icon";

const NavBar = styled.nav`
  background-color: rgba(0, 0, 0, 0.75);
  z-index: 1;
`;

const MenuOpen = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
`;

const MenuIcon = styled.span`
  font-size: 2rem;
  margin: 0.5rem;
`;

export const Navigation = ({
  menuInitialState = true,
  onMenuToggle,
  children
}) => {
  const [menuOpen, setMenuOpen] = useState(menuInitialState);

  return (
    <NavBar id="left-navigation" menuOpen={menuOpen}>
      <MenuOpen
        onClick={() => {
          setMenuOpen(!menuOpen);
          onMenuToggle(!menuOpen);
        }}
      >
        <Icon name={menuOpen ? "close" : "menu"} as={MenuIcon} />
      </MenuOpen>
      {children}
    </NavBar>
  );
};

Navigation.propTypes = {
  menuInitialState: PropTypes.bool,
  onMenuToggle: PropTypes.func,
  children: PropTypes.node
};

Navigation.defaultProps = {
  menuInitialState: true,
  onMenuToggle: () => null,
  children: ""
};
