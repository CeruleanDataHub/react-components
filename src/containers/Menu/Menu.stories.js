import React from "react";

import { Menu } from "./Menu.jsx";

export default {
  title: "Menu",
  component: Menu,
};

const items = [
  { name: "First", iconClass: "lnr-cog" },
  { name: "Second", iconClass: "lnr-eye" },
];

export const ChartStory = () => <Menu items={items} />;
