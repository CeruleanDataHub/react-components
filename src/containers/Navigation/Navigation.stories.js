import React, { useState } from "react";
import styled from "styled-components";

import { Menu } from "../Menu/Menu";
import { Navigation } from "./Navigation";

export default {
  title: "Navigation",
  component: Navigation
};

const menuItems = [
  { name: "First", icon: "cog" },
  { name: "Second", icon: "eye" }
];

export const NavigationStory = () => <Navigation />;

export const NavigationWithToggleStory = () => {
  const [menuOpen, setMenuOpen] = useState(true);
  return (
    <Navigation onMenuToggle={setMenuOpen}>
      <Menu items={menuItems} menuOpen={menuOpen} />
    </Navigation>
  );
};

const NavigationWrapper = styled.div`
  transition: width linear 280ms;
  background-color: rgba(0, 0, 0, 0.75);
  color: #ffffff;
  width: ${({ menuOpen }) => (menuOpen ? 20 : 4)}em;

  button {
    color: #ffffff;
  }

  .menu-text {
    transition: opacity linear 280ms;
    /* text still selectable in browser */
    opacity: ${({ menuOpen }) => (menuOpen ? 1 : 0)};
  }
`;

export const NavigationWithToggleAndAnimationStory = () => {
  const [menuOpen, setMenuOpen] = useState(true);

  return (
    <NavigationWrapper menuOpen={menuOpen}>
      <Navigation onMenuToggle={setMenuOpen} menuInitialState={menuOpen}>
        <Menu items={menuItems} menuOpen={menuOpen} />
      </Navigation>
    </NavigationWrapper>
  );
};

export const NavigationWithOpenMenuStory = () => (
  <Navigation>
    <Menu items={menuItems} menuOpen />
  </Navigation>
);

export const NavigationWithClosedMenuStory = () => (
  <Navigation menuInitialState={false}>
    <Menu items={menuItems} menuOpen={false} />
  </Navigation>
);
