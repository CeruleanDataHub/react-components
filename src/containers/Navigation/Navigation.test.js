import { mount } from "enzyme";
import PropTypes from "prop-types";
import React from "react";

import { Menu } from "../Menu/Menu";
import { Navigation } from "./Navigation";

const MenuLinkExample = ({ to, children }) => <a href={to}>{children}</a>;

MenuLinkExample.propTypes = {
  to: PropTypes.string,
  children: PropTypes.node
};

MenuLinkExample.defaultProps = {
  to: "",
  children: ""
};

const menuItems = [
  {
    name: "Home",
    icon: "home",
    LinkComponent: MenuLinkExample,
    to: "#home"
  },
  {
    name: "Users",
    icon: "users",
    LinkComponent: MenuLinkExample,
    to: "#users"
  },
  {
    name: "Devices",
    icon: "rocket",
    LinkComponent: MenuLinkExample,
    to: "#devices"
  },
  {
    name: "Costs",
    icon: "briefcase",
    LinkComponent: MenuLinkExample,
    to: "#costs"
  }
];

describe("Navigation", () => {
  it("renders", () => {
    const component = mount(<Navigation>Some children</Navigation>);

    expect(component).toMatchHtmlSnapshot();
  });

  it("given Menu as children, has Menu", () => {
    const component = mount(
      <Navigation>
        <Menu items={menuItems} />
      </Navigation>
    );

    expect(component.find("Menu")).toExist();
  });

  it("given menu initial state is false, has menu icon", () => {
    const component = mount(
      <Navigation menuInitialState={false}>
        <Menu items={menuItems} />
      </Navigation>
    );

    expect(component.find("Icon[data-icon-test]")).toHaveProp("name", "menu");
  });

  it("given onMenuToggle, does not call callback function yet", () => {
    const onMenuToggleMock = jest.fn();
    mount(
      <Navigation onMenuToggle={onMenuToggleMock}>
        <Menu items={menuItems} />
      </Navigation>
    );

    expect(onMenuToggleMock).not.toHaveBeenCalled();
  });

  it("given onMenuToggle, when called, calls callback function", () => {
    const onMenuToggleMock = jest.fn();
    const component = mount(
      <Navigation onMenuToggle={onMenuToggleMock}>
        <Menu items={menuItems} />
      </Navigation>
    );

    component.props().onMenuToggle();

    expect(onMenuToggleMock).toHaveBeenCalled();
  });

  it("should call default props onMenuToggle function when no onMenuToggle property is passed", () => {
    const component = mount(
      <Navigation>
        <Menu items={menuItems} />
      </Navigation>
    );

    const actual = component.props().onMenuToggle();

    expect(actual).toBeNull();
  });
});
