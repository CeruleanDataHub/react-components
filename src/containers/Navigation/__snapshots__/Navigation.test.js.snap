// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should call default props onMenuToggle function when no onMenuToggle property is passed 1`] = `
.c4 {
  text-align: center;
}

.c7 {
  font-family: "Exo 2",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c9 {
  font-size: 1.5em;
}

.c5 {
  margin: 1em 0 0 0.25em;
  padding: 0;
  list-style-type: none;
}

.c6 {
  display: block;
  padding: 1em;
  line-height: 1.5em;
  height: 1.5em;
  font-weight: 300;
  text-transform: uppercase;
}

.c6 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  margin-left: 1rem;
  opacity: 1;
}

.c2 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
  border: none;
  margin-left: 0.5em;
}

.c3 {
  font-size: 2rem;
  margin: 0.5rem;
}

<nav
  className="c0"
>
  <button
    className="c1 c2"
    color=""
    data-button-test={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="c3 c4 lni lni-close"
    />
  </button>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#home"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-home"
            />
            <span
              className="c10 menu-text"
            >
              Home
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#users"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-users"
            />
            <span
              className="c10 menu-text"
            >
              Users
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#devices"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-rocket"
            />
            <span
              className="c10 menu-text"
            >
              Devices
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#costs"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-briefcase"
            />
            <span
              className="c10 menu-text"
            >
              Costs
            </span>
          </div>
        </a>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Navigation should have menu toggle event listener 1`] = `
.c4 {
  text-align: center;
}

.c7 {
  font-family: "Exo 2",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c9 {
  font-size: 1.5em;
}

.c5 {
  margin: 1em 0 0 0.25em;
  padding: 0;
  list-style-type: none;
}

.c6 {
  display: block;
  padding: 1em;
  line-height: 1.5em;
  height: 1.5em;
  font-weight: 300;
  text-transform: uppercase;
}

.c6 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  margin-left: 1rem;
  opacity: 1;
}

.c2 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
  border: none;
  margin-left: 0.5em;
}

.c3 {
  font-size: 2rem;
  margin: 0.5rem;
}

<nav
  className="c0"
>
  <button
    className="c1 c2"
    color=""
    data-button-test={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="c3 c4 lni lni-close"
    />
  </button>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#home"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-home"
            />
            <span
              className="c10 menu-text"
            >
              Home
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#users"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-users"
            />
            <span
              className="c10 menu-text"
            >
              Users
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#devices"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-rocket"
            />
            <span
              className="c10 menu-text"
            >
              Devices
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#costs"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-briefcase"
            />
            <span
              className="c10 menu-text"
            >
              Costs
            </span>
          </div>
        </a>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Navigation should render 1`] = `
.c4 {
  text-align: center;
}

.c2 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
  border: none;
  margin-left: 0.5em;
}

.c3 {
  font-size: 2rem;
  margin: 0.5rem;
}

<nav
  className="c0"
>
  <button
    className="c1 c2"
    color=""
    data-button-test={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="c3 c4 lni lni-close"
    />
  </button>
  
</nav>
`;

exports[`Navigation should render children menu 1`] = `
.c4 {
  text-align: center;
}

.c7 {
  font-family: "Exo 2",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c9 {
  font-size: 1.5em;
}

.c5 {
  margin: 1em 0 0 0.25em;
  padding: 0;
  list-style-type: none;
}

.c6 {
  display: block;
  padding: 1em;
  line-height: 1.5em;
  height: 1.5em;
  font-weight: 300;
  text-transform: uppercase;
}

.c6 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  margin-left: 1rem;
  opacity: 1;
}

.c2 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
  border: none;
  margin-left: 0.5em;
}

.c3 {
  font-size: 2rem;
  margin: 0.5rem;
}

<nav
  className="c0"
>
  <button
    className="c1 c2"
    color=""
    data-button-test={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="c3 c4 lni lni-close"
    />
  </button>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#home"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-home"
            />
            <span
              className="c10 menu-text"
            >
              Home
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#users"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-users"
            />
            <span
              className="c10 menu-text"
            >
              Users
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#devices"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-rocket"
            />
            <span
              className="c10 menu-text"
            >
              Devices
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#costs"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-briefcase"
            />
            <span
              className="c10 menu-text"
            >
              Costs
            </span>
          </div>
        </a>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Navigation should render children menu closed when given appropriate property 1`] = `
.c4 {
  text-align: center;
}

.c7 {
  font-family: "Exo 2",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c9 {
  font-size: 1.5em;
}

.c5 {
  margin: 1em 0 0 0.25em;
  padding: 0;
  list-style-type: none;
}

.c6 {
  display: block;
  padding: 1em;
  line-height: 1.5em;
  height: 1.5em;
  font-weight: 300;
  text-transform: uppercase;
}

.c6 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  margin-left: 1rem;
  opacity: 1;
}

.c2 {
  font-family: inherit;
  background: transparent;
  font-size: 1em;
  padding: 0.25em 1.5em;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
}

.c0 {
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  height: 3.5em;
  cursor: pointer;
  border: none;
  margin-left: 0.5em;
}

.c3 {
  font-size: 2rem;
  margin: 0.5rem;
}

<nav
  className="c0"
>
  <button
    className="c1 c2"
    color=""
    data-button-test={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="c3 c4 lni lni-menu"
    />
  </button>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#home"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-home"
            />
            <span
              className="c10 menu-text"
            >
              Home
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#users"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-users"
            />
            <span
              className="c10 menu-text"
            >
              Users
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#devices"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-rocket"
            />
            <span
              className="c10 menu-text"
            >
              Devices
            </span>
          </div>
        </a>
      </div>
    </li>
    <li
      className="c6"
    >
      <div
        className="c7"
        fontFamily="exo"
        size="normal"
      >
        <a
          href="#costs"
        >
          <div
            className="c8"
          >
            <span
              className="c9 c4 lni lni-briefcase"
            />
            <span
              className="c10 menu-text"
            >
              Costs
            </span>
          </div>
        </a>
      </div>
    </li>
  </ul>
</nav>
`;
